files/files.go

```diff
@@ -359,3 +384,79 @@ func TempFileName(prefix, suffix string) string {
 	rand.Read(randBytes)
 	return filepath.Join(os.TempDir(), prefix+hex.EncodeToString(randBytes)+suffix)
 }
+
+//	Resolve File
+//
+//	If file location passed as argument is a valid,
+//	then it returns the contents of the file.
+//	Otherwise, an error.
+func ResolveFile(file string) (string, error) {
+
+	var payload string
+
+	//	Ensure it is a valid file path,
+	//	by ensuring there an extension suffixed at the end.
+	//
+	//	This will validate both a file in the filesystem,
+	//	as well as the URL source for a file.
+	if filepath.Ext(file) == "" {
+		return payload, ErrInvalidPath
+	}
+
+	//	Try to parse the file as a URL,
+	//	and go-get the file if it a valid URL.
+	//
+	//	For some reason, this inbuilt parser is broken
+	//	and it even matches relative paths that need not be URLs.
+	//	But we still need to parse the URL so that we can
+	//	validate the scheme to detect a URL.
+	url, err := url.ParseRequestURI(file)
+
+	//	Ensure the scheme is HTTP or HTTPS.
+	if err == nil &&
+		url.Scheme != "" &&
+		(strings.ToLower(url.Scheme) == "http" ||
```

Comment:
What about the other schemes supported by go-getter ?

Reply:
I only added the scheme check because FS files came with `file://` scheme. And I wanted to exclude them. Because the `url.ParseRequestURI(source)` method is parsing even the URLs without a host as valid ones. So, I needed a quick way out.

Reply:
Oh, wait! I didn't know that `go-getter` also supports local files. I thought it only supported remote files over networks.

This calls for changing the entire function code now.

Apologies for this mishap! I'll get this done.

Reply:
@moshloop, should I add an extra argument for the destination to where we have to resolve the file? The issue description doesn't mention it, that's why I am asking. In absence of a valid destination, I'll initialize a temporary directory to resolve the files over there.

Reply:
Pushed new changes. Supports all `go-getter` schemes now 👍 Including `git://`.